src/modules/facility-fixed-fee/dto/get-facility-fixed-fee-response.dto.ts

```diff
@@ -0,0 +1,100 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { EXAMPLES, PROPERTIES } from '@ukef/constants';
+import { DateOnlyString } from '@ukef/helpers';
+
+export type GetFacilityFixedFeeResponse = GetFacilityFixedFeeResponseItem[];
+export class GetFacilityFixedFeeResponseItem {
+  @ApiProperty({
+    description: 'The identifier of the facility.',
+    example: EXAMPLES.FACILITY_ID,
+    minLength: 10,
+    maxLength: 10,
+  })
+  readonly facilityIdentifier: string;
+
+  @ApiProperty({
+    description: 'The identifier of the portfolio.',
+    example: PROPERTIES.GLOBAL.portfolioIdentifier,
+  })
+  readonly portfolioIdentifier: string;
+
+  @ApiProperty({
+    description: 'The fixed amount to be billed to the client. If a flat amount is charged, this field is required.',
+    example: EXAMPLES.DEAL_OR_FACILITY_VALUE,
+  })
+  readonly amount: number;
+
+  @ApiProperty({
+    description: 'The effective date of this accruing/Fixed fee schedule.',
+    type: Date,
+    format: 'date',
+    example: '2023-03-24',
+  })
+  readonly effectiveDate: DateOnlyString;
+
+  @ApiProperty({
+    description: 'The expiration date of this accruing/Fixed fee schedule.',
+    type: Date,
+    format: 'date',
+    example: '2023-03-24',
+  })
+  readonly expirationDate: DateOnlyString;
+
+  @ApiProperty({
+    description: 'Date the next fee bill is due for this fee schedule.',
+    type: Date,
+    format: 'date',
+    example: '2023-03-24',
+  })
+  readonly nextDueDate: DateOnlyString;
+
+  @ApiProperty({
+    description: 'End date for the current accrual period. This date can be different than the Next Due Date.',
+    type: Date,
+    format: 'date',
+    example: '2023-03-24',
+  })
+  readonly nextAccrueToDate: DateOnlyString;
+
+  @ApiProperty({
+    description: 'Segment identifier from income exposure table. 2 alphanumeric characters.',
+    maxLength: 2,
+    minLength: 2,
```

Comment:
length:2 will be cleaner.

Reply:
updated to use `ValidatedStringApiProperty` to be able to use length